In [2]:
# importing all the imports needed for this phase
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import env

> ACQUIRE

In [6]:
# creating the url and creating SQL query to pull data
url = env.get_db_url('telco_churn')
query = "select * from customers join contract_types using(contract_type_id) join internet_service_types using(internet_service_type_id) join payment_types using(payment_type_id)"

In [7]:
# pulling database using read sql function
df = pd.read_sql(query, url)

In [8]:
# checking top 5 rows in data
df.head()in

payment_type_id  internet_service_type_id  contract_type_id customer_id  \
0                2                         1                 2  0002-ORFBO   
1                2                         1                 1  0003-MKNFE   
2                1                         2                 1  0004-TLHLJ   
3                1                         2                 1  0011-IGKFF   
4                2                         2                 1  0013-EXCHZ   

   gender  senior_citizen partner dependents  tenure phone_service  ...  \
0  Female               0     Yes        Yes       9           Yes  ...   
1    Male               0      No         No       9           Yes  ...   
2    Male               0      No         No       4           Yes  ...   
3    Male               1     Yes         No      13           Yes  ...   
4  Female               1     Yes         No       3           Yes  ...   

  tech_support streaming_tv streaming_movies paperless_billing  \
0          Yes          Yes               No               Yes   
1           No           No              Yes                No   
2           No           No               No               Yes   
3           No          Yes              Yes               Yes   
4          Yes          Yes               No               Yes   

  monthly_charges total_charges churn   contract_type  internet_service_type  \
0            65.6         593.3    No        One year                    DSL   
1            59.9         542.4    No  Month-to-month                    DSL   
2            73.9        280.85   Yes  Month-to-month            Fiber optic   
3            98.0       1237.85   Yes  Month-to-month            Fiber optic   
4            83.9         267.4   Yes  Month-to-month            Fiber optic   

       payment_type  
0      Mailed check  
1      Mailed check  
2  Electronic check  
3  Electronic check  
4      Mailed check  

[5 rows x 24 columns]

In [10]:
# checking the size of the data
df.shape

(7043, 24)

>there are 7043 rows and 24 columns

from this data i can initially see what are some columns that can be important and some that are not

important
> gender
> tenure
> senior
> monthly

unimportant
> payment_type_id
> internet_service_type
> contract_type_id
> customer_id ( theres already an index)  